### PR TITLE
Stack event handlers

### PR DESCRIPTION
**Pull request by jrmoulton**

### CODE REVIEW

File: src/views/decorator.rs [hunk 1]
```diff
@@ -96,41 +96,47 @@ pub trait Decorators: View + Sized {
     ///
     /// NOTE: View should have `.keyboard_navigable()` in order to receive keyboard events
     fn on_key_down(
```
Comment (Zoxc) [inline src/views/decorator.rs:98]: `on_key_down` and `on_key_up` are already stackable.

Comment (jrmoulton) [reply src/views/decorator.rs:98]: I removed the handlers from view data and made them just work like all of the other event handlers. Is there a reason to leave it this way?

Comment (Zoxc) [reply src/views/decorator.rs:98]: Yes. They have less startup cost as they directly mutate `ViewData` and don't need to send an update message.

Comment (jrmoulton) [reply src/views/decorator.rs:98]: Should any other events be implemented with handlers in ViewData?

File: src/context.rs [hunk 2]
```diff
@@ -903,21 +905,19 @@ impl<'a> EventCx<'a> {
             _ => (),
         }
 
-        for handler in &view.view_data().event_handlers {
```
Comment (Zoxc) [inline src/context.rs:906]: You should keep this handling. These are supposed to be unconditional and trigger once on any unhandled event.

File: src/views/decorator.rs (hunk 1 above)
Comment (Zoxc) [reply src/views/decorator.rs]: I'd move as many as possible to use `ViewData.event_handlers`. Just have to be careful to not change behavior. I'd probably add a `ViewData::add_event_handler` method which boxes and pushes the handler for convenience.

Keep that in a separate PR if you give it a shot.

**Pull request merged**